okhttp/src/main/java/io/grpc/transport/okhttp/AsyncFrameWriter.java

```diff
@@ -184,6 +189,10 @@ public void ping(final boolean ack, final int payload1, final int payload2) {
     executor.execute(new WriteRunnable() {
       @Override
       public void doRun() throws IOException {
+        if (frameWriter == null) {
+          // In case the upper layer calls ping when the connection hasn't been connected yet.
```

Comment:
This orphans the ping. It seems handling it like any other write would be fine, i.e., throw an exception.